src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -103,17 +103,24 @@ public String getScript() {
         scriptPath = shf.getRemote();
         List<String> args = new ArrayList<String>();
 
+        envVars.put(cookieVariable, "please-do-not-kill-me");
+        // The temporary variable is to ensure JENKINS_SERVER_COOKIE=durable-… does not appear even in argv[], lest it be confused with the environment.
+        Map<String, String> preparedVars = envVars;
+
         OsType os = ws.act(new getOsType());
 
         if (os != OsType.DARWIN) { // JENKINS-25848
             args.add("nohup");
         }
+
         if (os == OsType.WINDOWS) { // JENKINS-40255
             scriptPath= scriptPath.replace("\\", "/"); // cygwin sh understands mixed path  (ie : "c:/jenkins/workspace/script.sh" )
         }
+        else {
+            // Escaping creates issues when the connection is established through Cygwin sshd(see issue: JENKINS-41225)
```

Comment:
I wonder why escaping should matter between connection types. Jenkins should run as a native windows process regardless of connection type. Perhaps `PATH` or other variables differ when you launch through cygwin or not. Or are variables always broken when `bash` is from cygwin regardless of slave connection type.

Original issue [https://issues.jenkins-ci.org/browse/JENKINS-41225](JENKINS-41225) seems to be about failing to find `sh`. I wonder if it can have anything to do with an issue I just found where relative path for "Shell executable" == `bash` fails to work with pipeline step but works with free-style project shell build step. Maybe this is already solved in free-style shell step and same can be applied to the pipeline step.